dataRetriever/errors.go

```diff
@@ -13,6 +13,9 @@ var ErrNoConnectedPeerToSendRequest = errors.New("connected peers list is empty.
 // ErrNilValue signals the value is nil
 var ErrNilValue = errors.New("nil value")
 
+// ErrCannotFindTxInBlockPool signals the value is nil
+var ErrCannotFindTxInBlockPool = errors.New("cannot find tx in current block pool")
```

Comment:
might rename to `ErrTxNotFoundInBlockPool`

Reply:
renamed.